src/unix/pty.cc

```diff
@@ -242,85 +233,108 @@ NAN_METHOD(PtyFork) {
   sigfillset(&newmask);
   pthread_sigmask(SIG_SETMASK, &newmask, &oldmask);
 
-  pid_t pid = pty_forkpty(&master, nullptr, term, &winp);
+  int master, slave;
+  int ret = pty_openpty(&master, &slave, nullptr, NULL, &winp);
+  if (ret == -1) {
+    perror("openpty failed");
+    return Nan::ThrowError("openpty failed.");
+  }
 
-  if (!pid) {
-    // remove all signal handler from child
-    sig_action.sa_handler = SIG_DFL;
-    sig_action.sa_flags = 0;
-    sigemptyset(&sig_action.sa_mask);
-    for (int i = 0 ; i < NSIG ; i++) {    // NSIG is a macro for all signals + 1
-      sigaction(i, &sig_action, NULL);
-    }
+  int comms_pipe[2];
+  if (socketpair(AF_UNIX, SOCK_STREAM, 0, comms_pipe)) {
+    perror("socketpair failed");
+    return Nan::ThrowError("socketpair failed.");
   }
+
+  posix_spawn_file_actions_t acts;
+  posix_spawn_file_actions_init(&acts);
+  posix_spawn_file_actions_adddup2(&acts, slave, COMM_PTY_FD);
+  posix_spawn_file_actions_addclose(&acts, master);
+  posix_spawn_file_actions_addclose(&acts, slave);
+  posix_spawn_file_actions_adddup2(&acts, comms_pipe[1], COMM_PIPE_FD);
+  posix_spawn_file_actions_addclose(&acts, comms_pipe[0]);
```

Comment:
Would not do it that way here, but as you did it in the other PR:
- map slave to stdin/out/err
- map comm_pipe to stderr + 1

With that change you dont have to mess with the std FDs in the helper anymore, as they are correctly pointing to the pty slave already.